### ISSUE TITLE
Make crate::toggle specialized for interrupt configuration

### ISSUE THREAD

**Issue by Piroro-hs** [stm32-rs/stm32f3xx-hal#280]
Having one stop enum for all toggle-able things doesn't feel like a right abstraction.

(Yes, this issue should be opened before 0.8.0 release 😢)

**Comment by Sh3Rm4n** [comment]
Yeah. Maybe this will is the wrong abstraction. I'm fine with removing it in 0.9 again.
